docroot/profiles/custom/cgov_site/modules/custom/cgov_site_section/src/Plugin/Validation/Constraint/ValidatePrettyUrlConstraintValidator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Drupal\cgov_site_section\Plugin\Validation\Constraint;
+
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the ValidatePrettyUrlConstraintValidator constraint.
+ */
+class ValidatePrettyUrlConstraintValidator extends ConstraintValidator implements ContainerInjectionInterface {
+
+  /**
+   * Entity Type Manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('entity_type.manager'));
+  }
+
+  /**
+   * Constructs an ValidatePrettyUrlConstraintValidator object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager) {
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * Checks pretty url validation.
+   *
+   * @param mixed $field
+   *   The field/value that should be validated.
+   * @param \Symfony\Component\Validator\Constraint $constraint
+   *   The constraint for the validation.
+   */
+  public function validate($field, Constraint $constraint) {
+
+    if (!($constraint instanceof ValidatePrettyUrlConstraint)) {
+      return NULL;
+    }
+
+    $entity = $field->getEntity();
+    $prettyUrl = $entity->field_pretty_url->value;
+    if (!empty($prettyUrl)) {
+      $query = $this->entityTypeManager->getStorage('taxonomy_term')->getQuery()
+        ->condition('vid', $entity->bundle())
+        ->accessCheck(FALSE)
+        ->condition('field_pretty_url', $prettyUrl);
+      // Exclude the current term if it exists.
+      if (!$entity->isNew()) {
+        $query->condition('tid', $entity->id(), '!=');
+      }
+      $result = $query->execute();
+      if (!empty($result)) {
+        $this->context->addViolation($constraint->uniquePrettyUrl);
+      }
+    }
+  }
```

Comment:
Should this be returning something?  Line 51 above returns `NULL` if `$constraint` isn't the right type.

Reply:
It should not written anything, I have written same as in drupal core constraint validators.

Reply:
You are correct, it should not be returning anything.  Looking at how the `validate` method is declared in `Symfony\Component\Validator\ConstraintValidatorInterface`
([link](https://github.com/symfony/symfony/blob/v6.3.12/src/Symfony/Component/Validator/ConstraintValidatorInterface.php#L32)), the method is void and arguably it's the return NULL which should really just be `return`.